test/performance/scale_from_zero_test.go

```diff
@@ -129,48 +131,71 @@ func getStats(durations []time.Duration) *stats {
 	if len(durations) == 0 {
 		return nil
 	}
-	var avg time.Duration
+	var avg, min, max time.Duration
+	min = durations[0]
+	max = durations[0]
 
 	for _, dur := range durations {
+		if dur < min {
+			min = dur
+		}
+		if dur > max {
+			max = dur
+		}
 		avg += dur
 	}
 	avg = time.Duration(int64(avg) / int64(len(durations)))
 
 	return &stats{
 		avg: avg,
+		min: min,
+		max: max,
 	}
 }
 
 func testGrid(s *stats, tName string) error {
 	var tc []testgrid.TestCase
-	val := float32(s.avg.Seconds() / 1000)
-	tc = append(tc, CreatePerfTestCase(val, "Average", tName))
-	return testgrid.CreateTestgridXML(tc, tName)
+	avg := float32(s.avg.Seconds())
+	min := float32(s.min.Seconds())
+	max := float32(s.max.Seconds())
+	tc = append(tc, CreatePerfTestCase(avg, "Average", tName))
+	tc = append(tc, CreatePerfTestCase(min, "Minimum", tName))
+	tc = append(tc, CreatePerfTestCase(max, "Maximum", tName))
+	return testgrid.CreateTestgridXML(tc, "TestPerformanceScaleFromZero")
 }
 
-func testScaleFromZero(t *testing.T, count int) {
-	tName := fmt.Sprintf("TestScaleFromZero%d", count)
+func testScaleFromZero(t *testing.T, parallelism, numRuns int) {
+	runStats := make([]*stats, numRuns)
+	tName := fmt.Sprintf("TestScaleFromZero%d", parallelism)
 	logger := logging.GetContextLogger(tName)
-	durs, err := parallelScaleFromZero(logger, count)
-	if err != nil {
-		t.Fatal(err)
+
+	for i := 0; i < numRuns; i++ {
+		durs, err := parallelScaleFromZero(logger, parallelism)
+		if err != nil {
+			t.Fatal(err)
+		}
+		runStats[i] = getStats(durs)
+		logger.Infof("Test run %d: Average: %v", i, runStats[i].avg)
+	}
+
+	runDurs := make([]time.Duration, numRuns)
+	for i := 0; i < numRuns; i++ {
+		runDurs[i] = runStats[i].avg
 	}
-	stats := getStats(durs)
-	logger.Infof("Average: %v", stats.avg)
-	if err = testGrid(stats, tName); err != nil {
+	if err := testGrid(getStats(runDurs), tName); err != nil {
 		t.Fatalf("Creating testgrid output: %v", err)
 	}
 }
 
 func TestScaleFromZero1(t *testing.T) {
-	testScaleFromZero(t, 1)
+	testScaleFromZero(t, 1, 5)
```

Comment:
`(t, 1 /*scale*/, 5 /*runs*/)` same elsewhere? Usually helps with readability.

Reply:
or we can maybe make them const